### PR TITLE
Add chevron buttons to sub-items in the menu

### LINKED ISSUE #11043: Help menu sub-items cannot be access on a mobile browser
As reported by @ravenlune (9272ab2c-4b66-4b74-aebd-799725885d1c) and confirmed by me:

"On the mobile/small window version of the site, if you expand the menu and then tap on Help, you cannot access any of the links in the dropdown menu (the one that appears when hovering over it in a larger window) because it immediately redirects you to the static FAQ."

The same problem doesn't occur for the other menu items because even though they do redirect you to the menu item's top-level page, the menu remains open and you can still click on the sub-items under the top-level item.

You can replicate the bug on a PC by shrinking your browser window until the menu changes to the mobile/small-window version. It's not a significant problem on a PC because you can hover over any top-level item to get access to the sub-items, but hovering is not possible on a mobile device.

I'm going to leave this as suggestion-discussion for a few days in case the staff want to comment on how this would best be fixed but I'll mark it as help wanted next week if there's no objections to that.

This bug occurs on mobile device browsers.

### LINKED ISSUE #11155: For medium & small screen view menu bar is not hiding when clicked on a menu item.
[//]: # (Before logging this issue, please post to the Report a Bug guild from the Habitica website's Help menu. Most bugs can be handled quickly there. If a GitHub issue is needed, you will be advised of that by a moderator or staff member -- a player with a dark blue or purple name. It is recommended that you don't create a new issue unless advised to.)

[//]: # (Bugs in the mobile apps can also be reported there.)
For medium & small screen view menu bar is not hiding when clicked on a menu item.
[//]: # (If you have a feature request, use "Help > Request a Feature", not GitHub or the Report a Bug guild.)

[//]: # (For more guidelines see https://github.com/HabitRPG/habitica/issues/2760)

[//]: # (Fill out relevant information - UUID is found from the Habitia website at User Icon > Settings > API)
### General Info
  * UUID: 
  * Browser: Any
  * OS: 

### Description
[//]: # (Describe bug in detail here. Include screenshots if helpful.)
image

#### Console Errors
[//]: # (Include any JavaScript console errors here.)

### PR DESCRIPTION
**Pull request by HydeHunter2**
Fixes #11043
Fixes #11155

### Changes
- Add chevron buttons (Only in mobile version)
Github

----
UUID: af1477b6-509a-4611-b851-d7a8e3a9c02a

### CODE REVIEW

**Comment by HydeHunter2** [comment]
I don't know CSS well, so any comments and corrections are desirable

**Comment by Tressley** [comment]
The chevrons are too large in your screenshot. In Zeplin, focusing an object will display the properties in the right column. This should help a bit with CSS during implementation. 👍  

image

**Comment by HydeHunter2** [comment]
oops. Mistyped :(

My | Right
-------------|------------
width: ~~2~~4px; | width: 14px;

Now it looks like this:
Github
(Zeplin is on the right)

Thanks

**Comment by paglias** [comment]
@HydeHunter2 this is working well, thanks! Just one quick thing: after expanding a menu on mobile clicking on the chevron again doesn't close it. 

Anyway I'll put it up for testing on our test site

File: website/client/components/header/menu.vue [hunk 1]
```diff
@@ -18,37 +18,55 @@ div
...
+            .chevron-icon-up(v-html="icons.chevronUp")
+            .chevron-icon-down(v-html="icons.chevronDown")
           router-link.nav-link(:to="{name: 'items'}") {{ $t('inventory') }}
           .topbar-dropdown
             router-link.topbar-dropdown-item.dropdown-item(:to="{name: 'items'}", exact) {{ $t('items') }}
             router-link.topbar-dropdown-item.dropdown-item(:to="{name: 'equipment'}") {{ $t('equipment') }}
             router-link.topbar-dropdown-item.dropdown-item(:to="{name: 'stable'}") {{ $t('stable') }}
         li.topbar-item(:class="{'active': $route.path.startsWith('/shop')}")
+          .chevron-block
+            .chevron-icon-up(v-html="icons.chevronUp")
```
Comment (paglias) [inline website/client/components/header/menu.vue]: could you add a `v-once` to these elements (like icons) that never change?

**Review by paglias — COMMENTED** [review]
one quick perfomance tip :)

File: website/client/components/header/menu.vue (hunk 1 above)
Comment (HydeHunter2) [reply website/client/components/header/menu.vue]: Thanks for the tip :]
I correctly understood that I need to add a `v-once` _only_ to the icons?
Like this:
`.chevron-icon-up(v-html="icons.chevronUp", v-once)`

**Comment by HydeHunter2** [comment]
> @paglias -- ... Just one quick thing: after expanding a menu on mobile clicking on the chevron again doesn't close it.

Good. I'll fix this a little later.

File: website/client/components/header/menu.vue (hunk 1 above)
Comment (paglias) [reply website/client/components/header/menu.vue]: So v-once tells Vue not to render the element again which helps with performances. In that case icons seemed the only static thing but in general you can/should use it for every element that's not re-rendere again. For example

```
span {{ $t('someString') }}
```
never changes so it's fine to use v-once on it

**Comment by Tressley** [comment]
The design looks good, but I'd like to place an animation on the chevrons once we get that fix in. Here's a quick fiddle I threw together: https://jsfiddle.net/Tressley/L95adcqb/

Also, can we loop in #11155 ? I agree that the menu should close when navigating on small screens.

**Comment by HydeHunter2** [comment]
Tressley, Paglias -- Thank you so much for all these tips, comments and designs, but I'm a little busy right now and can only finish this PRs by the weekend.

I hope this doesn't bring down your plans or something else.

**Comment by paglias** [comment]
No worries @HydeHunter2 :) Take all the time you need!

**Comment by HydeHunter2** [comment]
Now chevrons are animated and you can close the tabs.

-----

I have 2 questions:
1) Can a user open all tabs?
Like this:
Github
Or after opening the tab should the previous tab be closed? (i.e. always open less than 2 tabs)

2) Do I need to animate dropdown list(`class="topbar-dropdown"`)?
If yes, could you help me? I haven't found how to animate the transition from `display: none` to `display: block`. It ignores animations and immediately takes on the final form.

Could you help?

-----

> @Tressley -- Also, can we loop in #11155?

Do you mean to add `Fixes #11155` to the description of PR?

-----

@paglias -- I'm not sure why commit fails the test.
There is an error about verifying payments credentials, but I only changed UI.
Could you take a look?

**Comment by paglias** [comment]
The test failure seems unrelated, I'll give it a look but nothing to worry about

**Comment by Tressley** [comment]
@HydeHunter2 -- Nice job! Let me see if I can provide some help.

1. A user shouldn't be able to open all tabs. It should work like an accordion component (opening the tab should close the previous tab)
2. I'm pretty sure you can't animate the `display` property, so my suggestion would be to use `height`

Also, this was a typo on my end, but could you update the transform property from `rotate(180deg)` to `rotate(-180deg)` (clockwise rotation)

As for #11155, I was asking if that was an issue you could combine with your current fixes. When you navigate on the mobile menu the menu stays expanded and can cause confusion. The fix is just to close the menu after navigating. Is that something you could include with these fixes? If not, it's okay, but I thought it might make sense as you're already in mobile menu land. 😄

**Comment by HydeHunter2** [comment]
- [x] `rotate(180deg)` -> `rotate(-180deg)`
- [x] Allow only 1 open tab
- [x] Animation to dropdown elements
- [x] Close menu after clicking tab (+`Fixes #11155` is added to the description of PR)

It seems everything is done :)

------

- [x] (?) Error about credentials

But there is still an error about `payments - stripe - #checkout ... verifies credentials` in the tests. And I have no idea what causes it.

File: website/client/components/header/menu.vue [hunk 2]
```diff
@@ -149,7 +169,35 @@ div
...
+        }
+
+        .chevron-icon-down {
+          width: 14px;
+          top: 11px;
+          right: 12px;
+          position: absolute;
+          display: block;
+          transition: transform 0.25s ease;
+     -moz-transition: transform 0.25s ease;
```
Comment (paglias) [inline website/client/components/header/menu.vue]: can you remove these browser prefixes? We have a plugin that automatically adds them if needed :) Same below with the `rotate` rule

File: website/client/components/header/menu.vue [hunk 3]
```diff
@@ -223,20 +271,28 @@ div
     font-weight: bold;
     transition: none;
 
-    .topbar-dropdown {
-      display: none; // Display is set to block on hover.
+    .topbar-dropdown  {
+        overflow: hidden;
+        max-height: 0;
+        transition: max-height 0.25s ease;
+   -moz-transition: max-height 0.25s ease;
```
Comment (paglias) [inline website/client/components/header/menu.vue]: here as well

**Review by paglias — CHANGES_REQUESTED** [review]
The test failure is unrelated, I left a comment about removing browser-specific css prefixes

**Comment by paglias** [comment]
I've fixed the test failure in the `develop` branch

**Comment by paglias** [comment]
@HydeHunter2 looks like there's a bug with the last version of the PR, in Chrome I'm not able to open the submenus by hovering the top menu label (for example hovering on Inventory doesn't show any submenu)

**Comment by HydeHunter2** [comment]
> ... open the submenus by hovering the top menu label ...

Oh, I thought it was necessary to replace opening by hovering with opening by clicking on chevrons.
(i.e. make clicking on chevrons the only way to open the menu)

-----
*These 2 commits don't fix this bug.

**Comment by paglias** [comment]
On mobile yes, the menus should open on click but on desktop the behavior should stay the same

**Comment by HydeHunter2** [comment]
Ahhhh. I see the bug you are talking about.

**Comment by HydeHunter2** [comment]
There is another bug in the navbar. Some tabs after clicking on them don't have an `active` class.
(i.e. light line at the bottom of the tab doesn't appear.)

It is | It should be
------|---------
Github | Github

`Tasks`, `Party`, `Guilds` tabs are buggy.
You can catch this bug in the `develop` branch and in the website.
Does it make sense to fix it in this PR?

**Comment by paglias** [comment]
Ah, I saw that but thought it was part of the same bug, but yeah it would be useful to fix it here, thanks!

**Comment by HydeHunter2** [comment]
Oh. It was harder than I thought :)
I had problems with `:class = "..."`. It sometimes behaves strangely.
Sometimes it overwrites other classes, and sometimes it does not. So in some places, I had to programmatically delete classes.

###### Do not pay attention to the last 2 commits. They don't matter.

**Comment by paglias** [comment]
thanks @HydeHunter2 ! It seems to be working well. I'll give a look at the code in the next few days to see if we can use more vue stuff but thanks a lot!

**Comment by Tressley** [comment]
@HydeHunter2 -- I seem to be having an issue with the "Groups" nav item not showing the dropdown. I tested on both desktop and mobile and while the dropdown items are present in the code, the dropdown itself is never visible. The other nav dropdowns seem to be working properly.

**Comment by Alys** [comment]
@HydeHunter2 In case you're not aware, when you're in guilds or a party that has a Group Plan, links to those groups appear under the Groups menu, so the menu items will be created dynamically for each player. If you want to test that locally, there's instructions for adding a Group Plan to a guild in https://habitica.fandom.com/wiki/Using_Your_Local_Install_to_Modify_Habitica%27s_Website_and_API

**Comment by paglias** [comment]
thanks @HydeHunter2 ! It seems to be working well now, a few more days of testing and then I think we can merge :)

**Comment by Tressley** [comment]
This LGTM! 👍

**Comment by paglias** [comment]
Then nevermind a few more days @HydeHunter2 :)

Merging this now and will go live on Habitica.com next week

**Comment by paglias** [comment]
@HydeHunter2 thanks a lot for this one! I've notated this towards your 4th tier, I haven't awarded you the 4th tier yet because the difficulty of each tier increases but we'll remember this was a significant contribution!

**Pull request merged**